nft/ethereum_payment_obligation.go

```diff
@@ -235,7 +235,11 @@ type MintRequest struct {
 
 // NewMintRequest converts the parameters and returns a struct with needed parameter for minting
 func NewMintRequest(to common.Address, anchorID anchors.AnchorID, proofs []*proofspb.Proof, rootHash [32]byte) (*MintRequest, error) {
-	tokenID := utils.ByteSliceToBigInt(utils.RandomSlice(256))
+
+	// tokenID is uint256 in Solidity (max value is 2^256-1)
+	// tokenID has 77 digits
+	// 1 byte = 8 bits
+	tokenID := utils.ByteSliceToBigInt(utils.RandomSlice(256/8))
```

Comment:
Why do we need to keep the `256` byte reference? If we know the type in solidity is uint256 (256 bits). Is there any reason why we cannot use `32` here?

Reply:
I found it a bit more confusing 😄 as the function name says: `ByteSliceToBigInt` already. Maybe `utils.RandomSlice` should be `utils.RandomByteSlice` instead.

Reply:
I changed it to 32 and added a better comment